### PR TITLE
update header and footer chrome

### PR DESCRIPTION
**Pull request by sandikbarr**
#109 and #110 

Click the spinner to toggle the userinfo loading/loaded. I did this so you could see the header's loading indicator in the styleguide demo.
screen shot 2018-06-04 at 5 07 05 pm

### CODE REVIEW

File: src/lib/assets/styles/variables.css [hunk 1]
```diff
@@ -1,5 +1,6 @@
 :root {
   --black: #000;
+  --grey-darkest: #1c1c1c;
```
Comment (sandikbarr) [inline src/lib/assets/styles/variables.css]: https://www.prx.org/company/brand-assets says `#1a1a1a` but uses `#1c1c1c` 🤔

File: src/lib/src/footer/footer.component.css [hunk 2]
```diff
@@ -2,12 +2,10 @@
  * PRX footer component
  */
 footer {
-  background: #fff;
+  background: #1c1c1c;
+  color: #bebebe;
```
Comment (sandikbarr) [inline src/lib/src/footer/footer.component.css]: Not sure on medium grey tone either... There is some kind of sorcery on the square space site that makes it look like these should be inheriting white from the footer, but they look grey to me.

File: src/lib/src/footer/footer.component.css [hunk 3]
```diff
@@ -24,22 +22,8 @@ footer {
  * Columns of footer content
  */
 .columns {
-  display: -webkit-box;
```
Comment (sandikbarr) [inline src/lib/src/footer/footer.component.css]: IE11 and up, so we're good on regular flex-box

File: src/lib/src/header/header.component.css [hunk 4]
```diff
@@ -1,55 +1,20 @@
...
   left: 0;
   width: 100%;
   max-width: 1200px;
   margin: 0 auto;
-  height: 50px;
   z-index: 200;
   border-bottom: 1px solid #e6e6e6;
-  background: #fff;
-}
-.contents {
```
Comment (sandikbarr) [inline src/lib/src/header/header.component.css]: dropped `.contents`

File: src/lib/src/header/header.component.css [hunk 5]
```diff
@@ -1,55 +1,20 @@
...
-  padding-left: 15px;
-}
-@media screen and (min-width: 768px) {
-  .contents {
-    padding-right: 30px;
-    padding-left: 30px;
-  }
-}
-a.drawer {
-  position: absolute;
```
Comment (sandikbarr) [inline src/lib/src/header/header.component.css]: none of this `.drawer` business was being used. I'm about to do some work on responsive design and would rather just start fresh with what we are using.

File: src/lib/src/header/header.component.css [hunk 6]
```diff
@@ -124,76 +61,6 @@ nav.alternate {
...
-  }
-  nav .nav-holder > *:not(.dropdown) {
-    line-height: 73px;
-  }
-}
-
-/*
- * Loading indicator
- */
-header.loading {
```
Comment (sandikbarr) [inline src/lib/src/header/header.component.css]: not used, the loading indicator is on the user image

File: src/demo/app/app.component.ts [hunk 7]
```diff
@@ -5,8 +5,9 @@ import { Component } from '@angular/core';
   template: `
     <prx-header>
       <prx-navitem route="/" text="PRX StyleGuide"></prx-navitem>
-      <prx-navuser userName="Mary">
-        <div class="user-loaded profile-image-placeholder"></div>
+      <prx-navuser [userinfo]="userinfo" (click)="toggleUserInfo()">
```
Comment (cavis) [inline src/demo/app/app.component.ts]: +1 nice for demoing.

File: README.md [hunk 8]
```diff
@@ -51,13 +51,12 @@ And add styles for any components you are using to your `.angular-cli.json`:
...
+To release to the npm registry, *make sure* you publish the `dist` folder.
 
 ```
-npm version patch
-npm run build
-cd dist/
-npm publish
+yarn version --patch
+yarn run build
+yarn publish dist
```
Comment (cavis) [inline README.md]: Handy!  i didn't know specifying a folder was even possible.  :+read-the-docs:

File: src/lib/src/auth/userinfo.service.ts [hunk 9]
```diff
@@ -29,7 +30,7 @@ export class UserinfoService {
 
   getUserinfo(): Observable<Userinfo> {
     const url = `${this.authHost}/userinfo`;
-    const params = new HttpParams().set('scope', 'profile+apps');
+    const params = new HttpParams().set('scope', 'profile+apps+email');
```
Comment (cavis) [inline src/lib/src/auth/userinfo.service.ts]: Hmm... any reason for adding email?  I think we usually display `firstname lastname`.  Not bad to have though.

File: src/lib/src/header/header.component.css (hunk 5 above)
Comment (cavis) [reply src/lib/src/header/header.component.css]: +1

File: src/lib/src/header/header.component.css (hunk 6 above)
Comment (cavis) [reply src/lib/src/header/header.component.css]: not actually an issue, but i notice the spinner isn't horizontally-centered on the image:

image

File: src/lib/src/footer/footer.component.css [hunk 10]
```diff
@@ -133,16 +110,7 @@ footer {
  * Social icons
  */
 ul.social {
-  display: -webkit-box;
```
Comment (cavis) [inline src/lib/src/footer/footer.component.css]: I'm guessing this was pre-existing, and will probably get fixed when mobile gets implemented ... but on smaller widths, the social icons get pretty close to the columns above:

image

File: README.md [hunk 11]
```diff
@@ -8,22 +8,22 @@ Get [Yarn](https://yarnpkg.com/en/docs/install) and then `yarn install`.
 
 ## Development server / Demo app
 
-Run `npm start` for a demo server. The app will automatically reload if you change any of the source files.
+Run `yarn start` for a demo server. The app will automatically reload if you change any of the source files.
```
Comment (cavis) [inline README.md]: Nobody can make me not type `npm start`.

File: src/lib/src/auth/userinfo.service.ts (hunk 9 above)
Comment (sandikbarr) [reply src/lib/src/auth/userinfo.service.ts]: yeah, email is what is displayed in the dropdown on corporate. I saw that username was removed from the header in the latest designs, so I assumed it was to move that user info into the menu as was done on corporate. We were previously displaying `name` from the userinfo response. (Personally, I think email can get a little long and stretch the menu out wide and maybe name was a better choice.)

File: src/lib/src/footer/footer.component.css (hunk 10 above)
Comment (sandikbarr) [reply src/lib/src/footer/footer.component.css]: I will definitely note this for the responsive work to be done on metrics. I also wondered if the social icons should move to the left and match the same set of corporate icons (same for columns of links.) The footers are all blank in the designs, so I don't think it has been thought out yet.

File: src/lib/src/header/header.component.css (hunk 6 above)
Comment (sandikbarr) [reply src/lib/src/header/header.component.css]: this positioning is fudged into place with negative margins and is an absolute nightmare :(

**Pull request merged**